hack/save-logs-files.sh

```diff
@@ -0,0 +1,234 @@
+}
+
+existant_ns=$(kubectl get ns -o=jsonpath='{.items[*].metadata.name}')
+eval set -- "$VALID_ARGS"
+while [ : ]; do
+    case "$1" in
+        -d | --duration)
+            duration=$2
+            # check that the argument is valid
+            re='^[0-9]+[smh]$'
+            if [[ ! $duration =~ $re ]]; then
+                echo "error: invalid argument '$duration' for -d|--duration flag: should contain a number and a time unit (s/m/h), e.g. 1h or 30m"
+                print_usage
+                exit 1
+            fi
+            shift 2
+            ;;
+        -n | --namespaces)
+            readarray -d $ARGS_SEPARATOR -t namespaces_list < <(printf '%s' "$2")
+            # check that all the given namespaces exist
+            for ns in ${namespaces_list[@]}; do
+                if [[ ! " ${existant_ns[@]} " =~ " ${ns} " ]]; then
+                    echo "error: given namespace '$ns' doesn't exist"
+                    print_usage
+                    exit 1
+                fi
+            done
+            shift 2
+            ;;
+        -p | --path)
+            output_path=$(readlink -f $2)/
+            if [[ ! -d $output_path ]]; then
+                echo "error: given path '$output_path' doesn't exist"
+                print_usage
+                exit 1
+            fi
+            shift 2
+            ;;
+        -a | --application)
+            application_args_list=()
+            readarray -d $ARGS_SEPARATOR -t application_args_list < <(printf '%s' "$2")
+            if [ ${#application_args_list[@]} -ne 2 ] || [ -z ${application_args_list[0]} ] || [ -z ${application_args_list[1]} ]; then
+                echo "error: given application namespace and name should be separated by a single comma"
+                print_usage
+                exit 1
+            fi
+            fybrik_application_ns=${application_args_list[0]}
+            fybrik_application_name=${application_args_list[1]}
+            if [[ ! " ${existant_ns[@]} " =~ " ${fybrik_application_ns} " ]]; then
+                echo "error: given namespace '$fybrik_application_ns' in flag '$FLAG_APP' doesn't exist"
+                print_usage
+                exit 1
+            fi
+            fybrik_applications_in_ns=$(kubectl get fybrikapplications -n $fybrik_application_ns -o=jsonpath='{.items[*].metadata.name}')
+            if [[ ! " ${fybrik_applications_in_ns[@]} " =~ " ${fybrik_application_name} " ]]; then
+                echo -e "error: given fybrikapplication '$fybrik_application_ns:$fybrik_application_name' doesn't exist"
+                print_usage
+                exit 1
+            fi
+            shift 2
+            ;;
+        -u | --uuid)
+            fybrik_application_uuid=$2
+            shift 2
+            ;;
+        -h | --help)
+            print_description
+            print_usage
+            print_flags
+            exit 0
+            ;;
+        --) shift; 
```

Comment:
I tried to run it on my Mac, the arguments parsing did not work.
the loop always break out here, and $1 equals to `d:n:p:u:a:h`.

Reply:
Looks like [in mac there is a different option of `getopt`](https://stackoverflow.com/a/7948533). It was decided that I will switch to use `getopts` which is included in `bash` but has only short flags.